Source/DafnyCore/Resolver/NameResolutionAndTypeInference/NameResolutionAndTypeInference.cs

```diff
@@ -319,15 +319,20 @@ public void ResolveAttributes(IAttributeBearingDeclaration attributeHost, Resolu
 
       // order does not matter much for resolution, so resolve them in reverse order
       foreach (var attr in attributeHost.Attributes.AsEnumerable()) {
-        if (attr is UserSuppliedAttributes) {
-          var usa = (UserSuppliedAttributes)attr;
+        if (attr is UserSuppliedAtAttribute { Builtin: true }) { // Already resolved
+          continue;
+        } else if (attr is UserSuppliedAttributes usa) {
           usa.Recognized = IsRecognizedAttribute(usa, attributeHost);
         }
         if (attr.Args != null) {
           foreach (var arg in attr.Args) {
             Contract.Assert(arg != null);
             if (!(Attributes.Contains(attributeHost.Attributes, "opaque_reveal") && attr.Name is "revealedFunction" && arg is NameSegment)) {
+              var prevCount = reporter.ErrorCount;
               ResolveExpression(arg, resolutionContext);
+              if (prevCount == reporter.ErrorCount && attr is UserSuppliedAtAttribute { Builtin: false }) {
+                reporter.Error(MessageSource.Resolver, attr.tok, "User-supplied @-Attributes not supported yet");
```

Comment:
Is it possible to hit this or is this just forwards-compatible defensive programming?

Reply:
It's not possible to hit this yet, but if someone added the ability to define custom attributes, it would be hit. I'll add this ability later and update this error reporting one, so it's forwards-compatible defensive programming, good guess! It decreases a bit code coverage but I think it's worth it. Also, it's to let you know where user-supplied @-attributes will be actually tested to have the @Attribute attribute.